Kill all qubes-guid processes on shutdown

marmarek opened the pull request:
Apparently those may stuck on shutdown (miss X server termination?),
delaying the whole system shutdown.

Fixes QubesOS/qubes-issues#1581

marmarek commented:
@adrelanos what do you think about this? Is that really such important - does leaking MAC is practical problem? Note that compromised VM can learn that it's running in Qubes using thousands different methods, so we're talking only about unintentional leaks here.
Alternatively, all VMs could use the same MAC address - somehow similar to https://github.com/QubesOS/qubes-issues/issues/1143. Since we use routing, not bridging, it is technically possible.

Having randomized MAC address I fear some usability issues, having Windows activation as an extreme example.

adrelanos commented:
This anonymizes MAC addresses within individual VMs, right? MAC randomizations for the "external" MAC adress from sys-net is [separate](https://www.qubes-os.org/doc/anonymizing-your-mac-address/)? I guess.

Difficult to say. In this case:

anonymity <-----> usability

Make your pick. If anonymity is more important than usability, then go for it.

I am not aware that any legitimate applications do read and leak MAC addresses yet on the Linux platform. (Even no malware doing this.) It's probably just a matter of time until some browser plugin is doing that.

I would like to have this feature on an opt-in basis. Able to choose this in qvm-prefs or even also in QVMM. And then enable it by default for all Whonix VMs. Together with #1143 that would provide added value.

marmarek committed:
Kill all qubes-guid processes on shutdown

Apparently those may stuck on shutdown (miss X server termination?),
delaying the whole system shutdown.

Fixes QubesOS/qubes-issues#1581
